src/xo-app/vm/tab-disks.js

```diff
@@ -1,146 +1,481 @@
+  isDragging: monitor.isDragging()
+}))
+@propTypes({
+  connectDragSource: propTypes.func.isRequired,
+  connectDropTarget: propTypes.func.isRequired,
+  index: propTypes.number.isRequired,
+  isDragging: propTypes.bool.isRequired,
+  id: propTypes.any.isRequired,
+  item: propTypes.object.isRequired,
+  move: propTypes.func.isRequired
+})
+class OrderItem extends Component {
+  _toggle = checked => {
+    const { item } = this.props
+    item.active = checked
+    this.forceUpdate()
+  }
+
+  render () {
+    const { item, connectDragSource, connectDropTarget } = this.props
+    return connectDragSource(connectDropTarget(
+      <li className='list-group-item'>
+        <Icon icon='grab' />
+        {' '}
+        <Icon icon='grab' />
+        {' '}
+        {item.text}
+        <span className='pull-right'>
+          <Toggle value={item.active} onChange={this._toggle} />
+        </span>
+      </li>
+    ))
+  }
+}
+
+@propTypes({
+  onClose: propTypes.func,
+  vm: propTypes.object.isRequired
+})
+@DragDropContext(HTML5Backend)
+class BootOrder extends Component {
+  constructor (props) {
+    super(props)
+    const { vm } = props
+    const order = this._parseBootOrder(vm.boot && vm.boot.order)
+    this.state = {order}
+  }
+
+  _parseBootOrder (bootOrder) {
+    const bootOptions = {
+      c: 'Hard-Drive',
+      d: 'DVD-Drive',
+      n: 'Network'
+    }
+    const order = []
+    if (bootOrder) {
+      for (let id of bootOrder) {
+        if (id in bootOptions) {
+          order.push({id, text: bootOptions[id], active: true})
+          delete bootOptions[id]
+        }
+      }
+    }
+    forEach(bootOptions, (text, id) => order.push({id, text, active: false}))
+    return order
+  }
+
+  moveOrderItem = (dragIndex, hoverIndex) => {
+    const order = this.state.order.slice()
+    const dragItem = order.splice(dragIndex, 1)
+    if (dragItem.length) {
+      order.splice(hoverIndex, 0, dragItem.pop())
+      this.setState({order})
+    }
+  }
+
+  _reset = () => {
+    const { vm } = this.props
+    const order = this._parseBootOrder(vm.boot && vm.boot.order)
+    this.setState({order})
+  }
+
+  _save = () => {
+    const { vm, onClose = noop } = this.props
+    const { order: newOrder } = this.state
+    let order = ''
+    forEach(newOrder, item => { item.active && (order += item.id) })
+    return setVmBootOrder(vm, order)
+      .then(onClose)
+  }
+
+  render () {
+    const { order } = this.state
+
+    return <form>
+      <ul>
+        {map(order, (item, index) => <OrderItem
+          key={index}
+          index={index}
+          id={item.id}
+          item={item}
+          move={this.moveOrderItem}
+        />)}
+      </ul>
+      <fieldset className='form-inline'>
+        <span className='pull-right'>
+          <ActionButton icon='save' btnStyle='primary' handler={this._save}>Save</ActionButton>
+          {' '}
+          <ActionButton icon='reset' btnStyle='default' handler={this._reset}>Reset</ActionButton>
```

Comment:
Is it ? I didn't know. In fact in bootstrap (for some versions I got used to at least), the default style is not the "default style if there is no style", it the name of a button skin, and I write it because it is the skin I choose. But if we say that in our components, it is a default style, it's ok ...